### PR TITLE
Makes ghost dojo mobs dust if they're dead and ghostless

### PR DESCRIPTION
**Pull request by Putnam3145**
## About The Pull Request

If a mob in the ghost dojo has no ghost to enter it and no person currently controlling it, it'll dust.

## Why It's Good For The Game

Free respawns have entirely foreseen consequences.

## Changelog
:cl:
tweak: Ghost dojo spawns will dust if their owner suicides or uses the "ghost" verb.
/:cl:

### CODE REVIEW

File: code/datums/elements/dusts_on_catatonia.dm [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+/datum/element/dusts_on_catatonia/Detach(mob/M)
+	. = ..()
+	if(M in attached_mobs)
+		attached_mobs -= M
+	if(!attached_mobs.len)
+		STOP_PROCESSING(SSprocessing,src)
+
+/datum/element/dusts_on_catatonia/process()
+	for(var/mob/M in attached_mobs)
```
Comment (Ghommie) [inline code/datums/elements/dusts_on_catatonia.dm]:
```
	for(var/A in attached_mobs)
		var/mob/M = A
```
It's a list made solely of mobs anyway. Can skip the typecheck.

**Review by Ghommie — APPROVED** [review]
Will conflict a little with the virtual reality component but otherwise all fine.

File: code/datums/elements/dusts_on_catatonia.dm (hunk 1 above)
Comment (Putnam3145) [reply code/datums/elements/dusts_on_catatonia.dm]: correct, weird oversight, thank you

**Comment by Ghommie** [comment]
More EOL troubles.

**Comment by Putnam3145** [comment]
Fixed

**Pull request merged**